c/secp256k1_keccak256_lock_tron.h

```diff
@@ -0,0 +1,47 @@
+ /* The script perform secp256k1_keccak256_sighash_all verification. */
+#define BLAKE2B_BLOCK_SIZE 32
+#define BLAKE160_SIZE 20
+#define PUBKEY_SIZE 65  // ETH address uncompress pub key 
+#define TEMP_SIZE 32768
+#define RECID_INDEX 64
+/* 32 KB */
+#define MAX_WITNESS_SIZE 32768
+#define SCRIPT_SIZE 32768
+#define SIGNATURE_SIZE 65
+
+#if (MAX_WITNESS_SIZE > TEMP_SIZE) || (SCRIPT_SIZE > TEMP_SIZE)
+#error "Temp buffer is not big enough!"
+#endif
+
+
+/*
+ * Arguments:
+ * ethereum address, keccak256 hash of pubkey last 20 bytes, used to
+ * shield the real pubkey.
+ *
+ * Witness:
+ * WitnessArgs with a signature in lock field used to present ownership.
+ */
+int verify_secp256k1_keccak_tron_sighash_all(unsigned char* message, unsigned char* eth_address, unsigned char* lock_bytes) {
```

Comment:
Many functions have implicit assumptions but they are not documented. For example, this function assumes that `message`, `eth_address` and `lock_bytes` are all fixed-length binary chunks.